packages/common/src/lib/core/utils/cli-utils.ts

```diff
@@ -0,0 +1,114 @@
+import { execSync, ExecSyncOptions } from 'child_process';
+import { ensureDirSync, existsSync, readJSONSync, rmSync, writeJsonSync } from 'fs-extra';
+import { dirname, join, relative, resolve } from 'path';
+
+export function createWorkspaceWithNxWrapper(name: string, pkgName: string, extraArgs='', useNxCloud=false, presetVersion='latest', silent = true){
+
+  const directory = resolve(process.cwd(), name);
+  rmSync(directory, {
+    recursive: true,
+    force: true,
+  });
+
+  ensureDirSync(directory)
+
+  execSync(`npx --yes nx@latest init --nxCloud=${useNxCloud} --useDotNxInstallation`, {
+    cwd: directory,
+    ...(silent ? { stdio: ['ignore', 'ignore', 'ignore'] } : {}),
+  });
+
+  const nxJsonPath = resolve(directory, 'nx.json');
+  const nxJson: { installation: { version: string; plugins: Record<string, string> } } = readJSONSync(nxJsonPath);
```

Comment:
You can safely use `import type` here to get ahold of NxJsonConfiguration from devkit, instead of needing a custom type